tests/LengthLimitedStreamTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+use Legionth\React\Http\LengthLimitedStream;
+use React\Stream\ReadableStream;
+
+class LengthLimitedStreamTest extends TestCase
+{
+    private $input;
+    private $stream;
+
+    public function setUp()
+    {
+        $this->input = new ReadableStream();
+    }
+
+    public function testSimpleChunk()
+    {
+        $stream = new LengthLimitedStream($this->input, 5);
+        $stream->on('data', $this->expectCallableOnceWith('hello'));
+        $stream->on('end', $this->expectCallableOnce());
+        $this->input->emit('data', array("hello world"));
+    }
+
+    public function testInputStreamKeepsEmitting()
+    {
+        $stream = new LengthLimitedStream($this->input, 5);
+        $stream->on('data', $this->expectCallableOnceWith('hello'));
+        $stream->on('end', $this->expectCallableOnce());
+
+        $this->input->emit('data', array("hello world"));
+        $this->input->emit('data', array("world"));
+        $this->input->emit('data', array("world"));
+    }
+
+    public function testZeroLength()
```

Comment:
1. Got it. The name is shit
2. What do you mean?
3. The `HttpServer` will handle these. As far I know I can't do much when no 'data' events are send.